pkg/controller/awsfederatedaccountaccess/awsfederatedaccountaccess_controller.go

```diff
@@ -550,3 +557,52 @@ func (r *ReconcileAWSFederatedAccountAccess) cleanFederatedRoles(reqLogger logr.
 
 	return nil
 }
+
+func (r *ReconcileAWSFederatedAccountAccess)  deleteNonAttachedCustomPolicy(reqLogger logr.Logger, awsClient awsclient.Client, federatedRoleCR *awsv1alpha1.AWSFederatedRole) error {
+
+	var policyMarker *string
+	// Paginate through custom policies
+	for {
+		policyListOutput, err := awsClient.ListPolicies(&iam.ListPoliciesInput{Scope: aws.String("Local"), Marker: policyMarker})
+		if err != nil {
+			if aerr, ok := err.(awserr.Error); ok {
+				switch aerr.Code() {
+				default:
+					reqLogger.Error(
+						aerr,
+						fmt.Sprintf(aerr.Error()),
+					)
+					reqLogger.Error(err, fmt.Sprintf("%v", err))
+					return err
+				}
+			}
+		}
+
+		for _, policy := range policyListOutput.Policies {
+			if *policy.PolicyName == federatedRoleCR.Spec.AWSCustomPolicy.Name {
+				_, err = awsClient.DeletePolicy(&iam.DeletePolicyInput{PolicyArn: policy.Arn})
+				if err != nil {
+					if aerr, ok := err.(awserr.Error); ok {
+						switch aerr.Code() {
+						default:
+							reqLogger.Error(
+								aerr,
+								fmt.Sprintf(aerr.Error()),
+							)
+							reqLogger.Error(err, fmt.Sprintf("%v", err))
+							return err
+						}
+					}
+				}
+			}
+		}
+
+		if *policyListOutput.IsTruncated {
+			policyMarker = policyListOutput.Marker
```

Comment:
I would put a continue in here to say if the output is truncated set the marker and continue. Then you can remove the else below and just have break.